skops/card/_model_card.py

```diff
@@ -1,61 +1,117 @@
 import os
 import re
 
-from modelcards import ModelCard
+from modelcards import CardData, ModelCard
 from sklearn.utils import estimator_html_repr
 
 import skops
 
 
-def _extract_estimator_config(model):
-    """Extracts estimator configuration and renders them into a vertical table.
-
-    Parameters
-    ----------
-        model (estimator): scikit-learn pipeline or model.
-
-    Returns
-    -------
-    str:
-        Markdown table of hyperparameters.
-    """
-    hyperparameter_dict = model.get_params(deep=True)
-    table = "| Hyperparameters | Value |\n| :-- | :-- |\n"
-    for hyperparameter, value in hyperparameter_dict.items():
-        table += f"| {hyperparameter} | {value} |\n"
-    return table
-
-
-def create_model_card(
-    model,
-    card_data,
-    **card_kwargs,
-):
-    """Creates a model card for the model and saves it to the target directory.
-
-    Parameters:
-    ----------
-    model: estimator
-        scikit-learn compatible estimator.
-    card_data: CardData
-        CardData object.
-    card_kwargs:
-        Card kwargs are information you can pass to fill in the sections of the
-        card template, e.g. model_description, citation_bibtex, get_started_code.
-    """
-    ROOT = skops.__path__
-    model_plot = re.sub(r"\n\s+", "", str(estimator_html_repr(model)))
-    hyperparameter_table = _extract_estimator_config(model)
-    card_data.library_name = "sklearn"
-    template_path = card_kwargs.get("template_path")
-    if template_path is None:
-        template_path = os.path.join(f"{ROOT[0]}", "card", "default_template.md")
-    card_kwargs["template_path"] = template_path
-    card = ModelCard.from_template(
-        card_data=card_data,
-        hyperparameter_table=hyperparameter_table,
-        model_plot=model_plot,
-        **card_kwargs,
-    )
-
-    return card
+class Card:
+    def __init__(self, model):
+        self.model = model
+        self.hyperparameter_table = self.extract_estimator_config()
+        self.model_plot = re.sub(r"\n\s+", "", str(estimator_html_repr(model)))
+        self.template_sections = {}
+        self.figure_paths = {}
+
+    def add(self, section, value):
+        """Takes values to fill model card template.
+        Parameters:
+        ----------
+        section: str
+                Section in the template.
+        value: str
+                Value to fill in the section.
+        """
+        self.template_sections[section] = value
+        return self
+
+    def add_inspection(self, plot_name, plot_path):
+        """Takes plots to fill model card template.
+        Parameters:
+        ----------
+        plot_path: str
+                Plot path as a string.
+        plot_name: str
+                Name of the plot.
+        """
+        self.template_sections[plot_name] = plot_path
```

Comment:
I would say you can add inspections in a separate attribute than `template_sections` so that bellow in `save` you also don't have to check if the template has a specific name matching this `plot_name` or not.

Reply:
You don't necessarily need to match plot, it just looks if template has anything that it used to not have, which is incredibly fragile btw so I'll apply your recommendation😅

Reply:
Addressed it.